Dockerfile

```diff
@@ -19,9 +19,22 @@ COPY Gemfile.lock Gemfile.lock
 
 COPY . .
 
+#######################################
+#Staging
+#######################################
+FROM dev AS staging
+
 RUN bundle install
 
 CMD ["sh", "entrypoint.sh"]
 
+#######################################
+#Production
+#######################################
+FROM dev AS prod
+
+ENV RAILS_ENV=production
 
+RUN bundle install
```

Comment:
You can actually skip bundling twice like this by using COPY --from=staging and copying the already bundled gems from the staging image to the produciton image. :)